pkg/model/resource/options.go

```diff
@@ -178,11 +180,11 @@ func (opts *Options) NewV1Resource(c *config.Config, doResource bool) *Resource
 	//  - In any other case, default to                          => project resource
 	// TODO: need to support '--resource-pkg-path' flag for specifying resourcePath
 	if !doResource {
-		file := filepath.Join("pkg", "apis", res.GroupPackageName, opts.Version,
-			fmt.Sprintf("%s_types.go", strings.ToLower(opts.Kind)))
+		file := replacer.Replace(filepath.Join(
+			"pkg", "apis", "%[group-package-name]s", "%[version]s", "%[kind]s_types.go"))
 		if _, err := os.Stat(file); os.IsNotExist(err) {
 			if coreDomain, found := coreGroups[opts.Group]; found {
-				pkg = path.Join("k8s.io", "api", opts.Group, opts.Version)
+				pkg = replacer.Replace(path.Join("k8s.io", "api", "%[group]s", "%[version]s"))
```

Comment:
Re-reading the code with fresh eyes, I realized, we introduced many new concepts while working through the iterations:

- replacer (strings.Replacer)
- Keys that can be replaced (kind, package-name, group etc...) which now contributors will have to dig through resource.go's Replace function to understand what all is supported.

I am not sure if the benefits are worth introducing this complexity. I know I should have spotted these in my earlier reviews, but didn't, so sorry about that.

One thing I realized, one thing is missing though: quick abstraction to `get-the-types-path-dir` to store `..._types.go` and `get-the-controller-path-dir` to store `..._controller.go`.

For example, in this function, we need `get-the-types-path-dir` in three places, it will help to have a helper just to do that.

Reply:
The helper issue has been solved, the complexity concern is being discussed.